csharp/src/Microsoft.ML.OnnxRuntime/DisposableNamedOnnxValue.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Numerics.Tensors;
+using System.Buffers;
+using System.Collections;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    public interface IDisposableReadOnlyCollection<T>: IReadOnlyCollection<T>, IDisposable
+    {
+
+    }
+
+    public class DisposableList<T> : List<T>, IDisposableReadOnlyCollection<T>
```

Comment:
Deriving from `List<T>` is not recommended - especially in a public class.

[https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections)

X DO NOT use ArrayList or List<T> in public APIs.